src/wpool_pool.erl

```diff
@@ -460,8 +460,23 @@ build_wpool(Name) ->
 next_wpool(Wpool) ->
   Wpool#wpool{next = (Wpool#wpool.next rem Wpool#wpool.size) + 1}.
 
--ifdef(r_18).
-  seed_random() -> random:seed(erlang:timestamp()).
--else.
-  seed_random() -> random:seed(now()).
--endif.
+rnd(Wpool_Size) ->
+  case application:get_env(worker_pool, random_fun) of
+    undefined ->
+      set_random_fun(),
+      rnd(Wpool_Size);
+    {ok, RndFun} ->
+      RndFun(Wpool_Size)
+  end.
+
+set_random_fun() ->
+  RndFun =
+    case code:ensure_loaded(rand) of
+      {module, rand} -> fun rand:uniform/1;
+      {error, _} ->
+        fun(Size) ->
+          _ = random:seed(os:timestamp()),
```

Comment:
You could easily seed this outside the fun declaration, specially because you are using it for load balancing, not cryptography.